### PR TITLE
QF-2744 Quickfix Admin Projects download latest version button

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
image

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-2744 Quickfix Admin Projects download latest version button](https://app.clickup.com/t/2192114/QF-2744)

File: docker-app/qfieldcloud/core/templates/admin/project_files_widget.html
Comment (faebebin) [inline docker-app/qfieldcloud/core/templates/admin/project_files_widget.html:238]: ... so I would let the rest ...

Comment (faebebin) [reply docker-app/qfieldcloud/core/templates/admin/project_files_widget.html:238]: and only fix the `downloadBtn.addEventListener` ...

**Review by faebebin — CHANGES_REQUESTED** [review]
see suggestion batch

File: docker-app/qfieldcloud/core/templates/admin/project_files_widget.html
Comment (faebebin) [inline docker-app/qfieldcloud/core/templates/admin/project_files_widget.html:220]:
```suggestion
```

File: docker-app/qfieldcloud/core/templates/admin/project_files_widget.html
Comment (faebebin) [inline docker-app/qfieldcloud/core/templates/admin/project_files_widget.html:238]:
```suggestion
```

File: docker-app/qfieldcloud/core/templates/admin/project_files_widget.html
Comment (why-not-try-calmer) [reply docker-app/qfieldcloud/core/templates/admin/project_files_widget.html:220]: If nothing is selected, does it really make sense to let `info` or `delete` work?

**Comment by why-not-try-calmer** [comment]
Actually, there is no need at all for a version selector, is there? After all, `Delete` and `Info` apply to the filename, and `Download` could simply get the value of the latest version from the JSON API, without the need to read any value from the form.

**Comment by faebebin** [comment]
Ok, I do not know much about the file versioning actually. But then I would just to fix the current feature (which IMO my suggestion would do?) And then take it back to the support to ask what they really need before modifying the feature? Maybe comment your suggestions into clickup for discussion with support team

**Comment by why-not-try-calmer** [comment]
> Ok, I do not know much about the file versioning actually. But then I would just to fix the current feature (which IMO my suggestion would do?) And then take it back to the support to ask what they really need before modifying the feature? Maybe comment your suggestions into clickup for discussion with support team

Yes, the selector only matter for downloads, and is therefore a bit misleading. It would make sense if we allowed also deletion to target specific files, so I'll forward this question.

I have reduced the scope of this PR to just fixing what needed to, relabeling here and there. Feel free to approve if you have no more objections!

**Comment by faebebin** [comment]
@why-not-try-calmer did you ever try the following: Select a version > click 'delete file' . For me, it only deletes the chosen version? Should be master state I guess ...

**Comment by faebebin** [comment]
Further the info button indeed shows info for all versions, so it could IMO be replaced on the right and maybe with a lilttle seperator to clarify that does not relate to the selected option.

**Comment by why-not-try-calmer** [comment]
> @why-not-try-calmer did you ever try the following: Select a version > click 'delete file' . For me, it only deletes the chosen version? Should be master state I guess ...

No you are correct. I didn't expect the file version to be specified as part of the request's headers. I changed the html text of the Download button accordingly.

**Comment by why-not-try-calmer** [comment]
^ updated screenshot in first commit message.

**Comment by faebebin** [comment]
@why-not-try-calmer on master, if the default was selected and then clicked 'delete' then all versions (whole file) was deleted. So the title of 'delete version' button would be wrong in this case. And we do not want to change the logic of the default option. It should be a bugfix not new feature. IMO the feature was working with this one bug described in the task.

Please just implement my suggestion of just fixing downloadBtn.addEventListener https://github.com/opengisch/qfieldcloud/pull/708 . (In your version you do not take into account if there are no versions which IMO could be the case with some special files. At least I don't know and did not test it, so I just let this logic.) Further I don't know whether the sorting is necessary but maybe I am wrong there - then please correct me with specific code.

But anyway - let's not make a feature-PR out of this but stay with a bugfix. We should not lose more time on this. Else feel free to pass over to me. I won't approve it as a feature change.

**Comment by why-not-try-calmer** [comment]
> @why-not-try-calmer on master, if the default was selected and then clicked 'delete' then all versions (whole file) was deleted. So the title of 'delete version' button would be wrong in this case. And we do not want to change the logic of the default option. It should be a bugfix not new feature. IMO the feature was working with this one bug described in the task.
> 
> Please just implement my suggestion of just fixing downloadBtn.addEventListener #708 . (In your version you do not take into account if there are no versions which IMO could be the case with some special files. At least I don't know and did not test it, so I just let this logic.) Further I don't know whether the sorting is necessary but maybe I am wrong there - then please correct me with specific code.
> 
> But anyway - let's not make a feature-PR out of this but stay with a bugfix. We should not lose more time on this. Else feel free to pass over to me. I won't approve it as a feature change.

Thanks for the suggestion, I think we are all good now. Can you checkout, test and confim?

**Comment by suricactus** [comment]
Duplicate of https://github.com/opengisch/qfieldcloud/pull/708 .
Looking at the amount of changes here, it seems the PR went out of scope according to the original task on CU.
There are still some unresolved comments which probably cause this PR not to be mergeable yet.

Since we have an alternative implementation, I am closing this for now.

**Comment by suricactus** [comment]
Ok, we did another round of reviewing with @why-not-try-calmer . We both agreed visual changes are best to be pushed in a separate PR and keep the bug fix PRs with less changes and therefore dicussion. For the future we will try to align on this.

@suricactus comment here:
The actual JS bugfix here is slightly more rubust and future proof because it does not depend on the default sorting that comes from the template (within the `<select><option>`) but really sorts the versions and makes sure we pick the most recent one.

@faebebin I hope you are ok with the current state of the PR. If yes, please close your PR here  https://github.com/opengisch/qfieldcloud/pull/708 and we keep the current state of this PR.

**Review by suricactus — APPROVED** [review]

**Pull request merged**
